### PR TITLE
chore(github-growth): track programming languages per repo

### PR DESCRIPTION
**Pull request by cathteng**
Populate the languages column in the Repository model (as introduced in #56751) when we consume Github commit push webhooks.

### CODE REVIEW

File: src/sentry/integrations/github/webhook.py [hunk 1]
```diff
@@ -47,6 +47,18 @@ def get_github_external_id(event: Mapping[str, Any], host: str | None = None) ->
     return f"{host}:{external_id}" if host else external_id
 
 
+def get_file_language(filename: str):
```
Comment (schew2381) [inline src/sentry/integrations/github/webhook.py:50]: imagine not typing lul
```suggestion
def get_file_language(filename: str) -> Union[str, None]:
```

File: src/sentry/integrations/github/webhook.py [hunk 2]
```diff
@@ -356,26 +369,36 @@ def _handle(
...
                             organization_id=organization.id,
                             commit=c,
                             filename=fname,
                             type="M",
                         )
+
+                    languages.discard(None)
+                    repo_languages = set(repo.languages)
+                    repo_languages.update(languages)
+                    repo.languages = list(repo_languages)
```
Comment (schew2381) [inline src/sentry/integrations/github/webhook.py:399]: maybe make this shorter?
```suggestion
                   repo.languages = list(set(repo.languages).union(languages))
```

Also if the db value for languages is initially null, `set(repo.languages)` will error. Is this a concern?

File: tests/sentry/integrations/github/test_webhooks.py
Comment (schew2381) [inline tests/sentry/integrations/github/test_webhooks.py:1]: Can you make the following tests:
1. unknown file language
2. multiple files of same type (i.e. added) w/ different languages 
3. Three different language for added, removed, and modified file types

File: src/sentry/integrations/github/webhook.py (hunk 1 above)
Comment (cathteng) [reply src/sentry/integrations/github/webhook.py:50]: typing check didn't complain 👀

File: tests/sentry/integrations/github/test_webhooks.py
Comment (cathteng) [reply tests/sentry/integrations/github/test_webhooks.py:1]: in `PUSH_EVENT_EXAMPLE_INSTALLATION` we have an unknown file language and a file with a known language in the same push event (over multiple commits), but both have been "modified"

modified this to have at least one entry in added + removed + modified over all the commits in the event. also added a file with another language

**Review by schew2381 — APPROVED** [review]
lgtm

File: src/sentry/integrations/github/webhook.py (hunk 1 above)
Comment (JoshFerge) [reply src/sentry/integrations/github/webhook.py]: on new files our typing is not set to strict -- but nice stuff. I think `str | None` is slightly preferred.

**Pull request merged**